pkg/k8sclient/k8sclient.go

```diff
@@ -62,8 +62,14 @@ func GetNodeByLabel(label string, selectorMethod string, rr *roundrobin.Balancer
 		return nodeList.Items[0], nil
 	default:
 		if selectorMethod == RR {
-			nodeId, _ := rr.Pick()
-			returnNode = nodeList.Items[nodeId.(int)%len(nodeList.Items)]
+			for range nodeList.Items {
```

Comment:
why not create the initial list from the nodes pre-selected, and then simply continue RRing from that list
you might even be able to off-load this selection to the client query, depending on what is possible to be configured into "listOptions"